### DISCUSSION TITLE
How to convert Amazon.DynamoDBv2.DocumentModel.Document to Dictionary<string, Amazon.Lambda.DynamoDBEvents.DynamoDBEvent.AttributeValue> ?

### DISCUSSION THREAD

**Discussion by Trisped** [aws/aws-lambda-dotnet#1675] [Q&A] [answered]
To simplify dev testing of stream processors on their system, we convert a JSON string to a Amazon.DynamoDBv2.DocumentModel.Document, then call the ToAttributeMap() method to convert it to a Dictionary<string, Amazon.Lambda.DynamoDBEvents.DynamoDBEvent.AttributeValue> and assign it to the image:

```C#
var json = @"{
	""rk"": ""rk value"",
	""pid"": ""pid guid"",
	...
}";
var record = JsonConvert.DeserializeObject<Dictionary<string, DynamoDBEntry>>(json);
var recordDocument = new Document(record);

var dynamoDbEvent = new DynamoDBEvent()
{
	Records = new List<DynamodbStreamRecord>()
	{
		new DynamodbStreamRecord()
		{
			Dynamodb = new StreamRecord()
			{
				NewImage = recordDocument.ToAttributeMap()
			}
		}
	}
};
```

This no longer works after updating to Amazon.Lambda.DynamoDBEvents (3.0.0).

Can someone direct me on the new way of doing this?

This has some similarities to discussion [1654](https://github.com/aws/aws-lambda-dotnet/discussions/1654) and issue [1657](https://github.com/aws/aws-lambda-dotnet/issues/1657).

**Comment by ashishdhingra** [comment]
@Trisped Good afternoon. I think your scenario is similar to what is described in https://github.com/aws/aws-lambda-dotnet/issues/1700 (this is reviewed with the team and in our backlog). Kindly confirm.

**Reply by Trisped** [reply]
@ashishdhingra Yes, #1700 looks the same as my issue.  I expect if #1700 is resolved, I will be able to use a similar solution in my code.

**Reply by ashishdhingra** [reply, marked as answer]
@Trisped Thanks. I will mark this discussion as answered since we have tracking feature request https://github.com/aws/aws-lambda-dotnet/issues/1700.

**Reply by Trisped** [reply]
@ashishdhingra I don't know the rules of this GitHub project, but wouldn't an answer require a solution to the problem?  As it is now, they is only a feature request; I don't have a way to convert my JSON to `Dictionary<string, DynamoDBEvent.AttributeValue>`.  If someone with the same issue googles this, they also won't be able to fix their issue.
